backend/integrations/oauth/oauthResource.go

```diff
@@ -0,0 +1,61 @@
+package oauth
+
+import "github.com/GenerateNU/sac/backend/config"
+
+type OAuthResourceClientInterface interface {
+	AuthorizeURL(state string) string
+	GetConfig() config.OAuthConfig
+}
+
+type ResourceClient struct {
+	Outlook 	OutlookOAuthClient
+	Google 		GoogleOAuthClient
+}
+
+// Calendar Types: 
+type OutlookOAuthClient struct {
+	OAuthConfig config.OAuthConfig
+}
+type GoogleOAuthClient struct {
+	OAuthConfig config.OAuthConfig
+}
+
+/**
+* GOOGLE OAUTH CLIENT
+**/
+func (client GoogleOAuthClient) AuthorizeURL(state string) string {
+	return (
+		client.OAuthConfig.BaseURL + "/auth?" + 
```

Comment:
there might be a url builder utility we can use here. if not, we should probably prefer fmt.Sprintf over string concat